testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -172,12 +172,12 @@
 end
 
 When(/^I select "([^"]*)" from "([^"]*)"$/) do |option, field|
-  xpath_option = ".//*[contains(@class, 'data-testid-#{field}-child__option') and contains(text(),'#{option}')]"
-  xpath_field = "//*[contains(@class, 'data-testid-#{field}-child__control')]"
   if has_select?(field, with_options: [option], wait: 1)
     select(option, from: field)
   else
     # Custom React selector
+    xpath_field = "//*[contains(@class, 'data-testid-#{field}-child__control')]"
+    xpath_option = ".//*[contains(@class, 'data-testid-#{field}-child__option') and contains(text(), '#{option}')]"
     find(:xpath, xpath_field).click
     find(:xpath, xpath_option, match: :first).click
```

Comment:
```suggestion
    xpath_field = "//*[contains(@class, 'data-testid-#{field}-child__control')]"
    xpath_option = ".//*[contains(text(), '#{option}')]"
    find(:xpath, xpath_field).click
    first(:xpath, xpath_option).click
```

Reply:
That's dangerous, this will not be searched within the field, but in the whole HTML page. Checking for `'data-testid-#{field}-child__option'` guarantees we will match only this field option.

I tried to keep the changes minimal in order to not introduce further problems.

Reply:
It's even worse then. Same answer applies.

Reply:
Anyway, if it worked, merge it, all right.

Reply:
Yes, I would have preferred using a `within`, it's good prectice.

However it would not work here: ReactJS does not store the options of the combobox within the combobox, but separately, at the end of the HTML page!

It nearly drove me mad until I understood it. I spent hours trying to understand why my `within` was not working :smiling_imp: .